Cleanup (#385)
theforeman/foreman-ansible-modules: README.md

When looking at actual modules in this repository ([`foreman_domain`](plugins/modules/foreman_domain.py) is a nice short example), you will notice a few differences to a "regular" Ansible module:

* Instead of `AnsibleModule`, we use `ForemanEntityApypieAnsibleModule` (and a few others, see [`plugins/module_utils/foreman_helper.py`](plugins/module_utils/foreman_helper.py)) which provides an abstraction layer for talking with the Foreman API

> * Instead of `argument_spec`, we provide an enhanced version called `entity_spec`. It handles the translation from module parameters to Foreman API parameters, as nobody wants to write `organization_ids` in their playbook when they can write `organizations`

Reviewer: ```suggestion
* Instead of Ansible's `argument_spec`, we provide an enhanced version called `entity_spec`. It handles the translation from Ansible module arguments to Foreman API parameters, as nobody wants to write `organization_ids` in their playbook when they can write `organizations`
```

Please note: we currently have modules that use `apypie` and `nailgun` as the backend libraries to talk to the API, but we would prefer not to add any new modules using `nailgun` and focus on migrating everything to `apypie`.

### Specification of the `entity_spec`

> In addition to Ansible's `argument_spec`, `entity_spec` understands the following types:

Reviewer: ```suggestion
The `entity_spec` can be seen as an extended version of Ansible's `argument_spec`. It understands more parameters (e.g. `flat_name`) and supports more `type`s than the original version. An `argument_spec` will be generated from a `entity_spec`, any parameters not directly known or consumed by `entity_spec` will be passed directly to the `argument_spec`.

In addition to Ansible's `argument_spec`, `entity_spec` understands the following types:
```

> * `type='entity'` The referenced value is another foreman entity.

Reviewer: ```suggestion
* `type='entity'` The referenced value is another Foreman entity.
```

This is usually combined with `flat_name=<entity>_id`.

> * `type='entity_list'` The referenced value is a list of foreman entities.

Reviewer: ```suggestion
* `type='entity_list'` The referenced value is a list of Foreman entities.
```

This is usually combined with `flat_name=<entity>_ids`.

> * `type='nested_list'` The referenced value is a list of foreman entities that are not included in the main api call.

Reviewer: ```suggestion
* `type='nested_list'` The referenced value is a list of Foreman entities that are not included in the main API call.
```

The module must handle the entities separately.
See domain parameters for an example.
The sub entities must be described by `entity_spec=<sub_entity>_spec`.

> * `type='invisible'` The parameter is avaliable to the api call, but it will be excluded from ansibles `argument_spec`.

Reviewer: ```suggestion
* `type='invisible'` The parameter is available to the API call, but it will be excluded from Ansible's `argument_spec`.
```

> `flat_name` provides a way to translate the name of a parameter as known to Ansible to the name understood by the foreman api.

Reviewer: ```suggestion
`flat_name` provides a way to translate the name of a module argument as known to Ansible to the name understood by the Foreman API.
```

> You can add or override ansible module parameters, by specifying them in `argument_spec` as usual.

Reviewer: ```suggestion
You can add new or override generated Ansible module arguments, by specifying them in the `argument_spec` as usual.
```

Please note: we currently have modules that use `apypie` and `nailgun` as the backend libraries to talk to the API, but we would prefer not to add any new modules using `nailgun` and focus on migrating everything to `apypie`.

### Specification of the `entity_spec`

> The `entity_spec` can be seen as an extended version of Ansible's `argument_spec`. It understands more parameters (e.g. `flat_name`) and supports more `type`s than the original version. An `argument_spec` will be generated from a `entity_spec`, any parameters not directly known or consumed by `entity_spec` will be passed directly to the `argument_spec`.

Reviewer 2: ```suggestion
The `entity_spec` can be seen as an extended version of Ansible's `argument_spec`. It understands more parameters (e.g. `flat_name`) and supports more `type`s than the original version. An `argument_spec` will be generated from an `entity_spec`. Any parameters not directly known or consumed by `entity_spec` will be passed directly to the `argument_spec`.
```

In addition to Ansible's `argument_spec`, `entity_spec` understands the following types:

* `type='entity'` The referenced value is another Foreman entity.
This is usually combined with `flat_name=<entity>_id`.
* `type='entity_list'` The referenced value is a list of Foreman entities.
This is usually combined with `flat_name=<entity>_ids`.
* `type='nested_list'` The referenced value is a list of Foreman entities that are not included in the main API call.
The module must handle the entities separately.

> See domain parameters for an example.

Reviewer 2: Perhaps domain could be a link? Otherwise make it clearer that this is the domain entity?
